File: app/src/main/java/com/tronography/pixelweather/dagger/AppComponent.java
package com.tronography.pixelweather.dagger;

import com.tronography.pixelweather.weather.WeatherActivity;

import javax.inject.Singleton;

import dagger.Component;


@Singleton
@Component(modules = {AppModule.class, NetworkModule.class, SharedPrefModule.class})
public interface AppComponent {

    void inject(WeatherActivity target);
}

File: app/src/main/java/com/tronography/pixelweather/dagger/AppComponent.kt
package com.tronography.pixelweather.dagger

import com.tronography.pixelweather.weather.WeatherActivity

import javax.inject.Singleton

import dagger.Component


@Singleton
@Component(modules = arrayOf(AppModule::class, NetworkModule::class, SharedPrefModule::class))
interface AppComponent {

    fun inject(target: WeatherActivity)
}
